src/shared/scrapers/US/TX/harris-county.js

```diff
@@ -0,0 +1,146 @@
+const getDataFromPivotTable = $ => {
+  // check that we have Date
+  const indexNameVal = $('div.corner > div > div.pivotTableCellWrap')
+    .text()
+    .trim();
+  if (indexNameVal.trim() !== 'Date') {
+    throw Error(`Expected index name to be Date, found ${indexNameVal}`);
+  }
+
+  const colnames = _getTableVals($, 'columnHeaders').slice(0, 2);
+  const expectedColname = [
+    'Patients in General Beds (Suspected + Confirmed)',
+    'Patients in Intensive Care Beds (Suspected + Confirmed)'
+  ];
+  assert.equal(colnames.join(';'), expectedColname.join(';'), 'Column names');
+
+  const dates = _getTableVals($, 'rowHeaders')
+    .map(x => new Date(x))
+    .map(datetime.parse)
+    .map(x => x.toString());
+
+  const ret = {
+    colnames: _getTableVals($, 'columnHeaders'),
+    dates,
+    generalBeds: _getColumnData($, 0),
+    icuBeds: _getColumnData($, 1)
+  };
+
+  assert.equal(ret.dates.length, ret.generalBeds.length, 'general beds');
+  assert.equal(ret.dates.length, ret.icuBeds.length, 'icu beds');
+  return ret;
+};
+
+const scraper = {
+  county: 'fips:48201',
+  state: 'iso2:US-TX',
+  country: 'iso1:US',
+  url:
+    'https://app.powerbi.com/view?r=eyJrIjoiYjU5MzU4NjAtZWJjMC00MTllLTkwYjYtMzE4ODY1YjAyMGU2IiwidCI6ImI3MjgwODdjLTgwZTgtNGQzMS04YjZmLTdlMGUzYmUxMGUwOCIsImMiOjN9',
+  sources: [
+    {
+      name: 'SouthEast Texas Regional Advisory Council',
+      url: 'https://www.setrac.org/',
+      description: 'SouthEast Texas Regional Advisory Council'
+    }
+  ],
+  _counties: ['Harris County'],
+  timeseries: true,
+  aggregate: 'county',
+  type: 'json',
+  maintainers: [maintainers.sglyon],
+  scraper: {
+    '0': async function() {
+      const callback = async page => {
+        page.setDefaultTimeout(10000);
+        const icuButton = await page.waitForXPath("//span[text()='Hospital/COVID Census']/..");
+        icuButton.click();
+
+        const harrisSelector = await page.waitForXPath("//div[@aria-label='Harris']");
+        harrisSelector.click();
+
+        // use context menu to open table version of chart
+        await (await page.$$('div.visual-lineChart'))[0].click({ button: 'right' });
+        const tableButton = await page.waitForXPath("//drop-down-list-item[//h6[text()='Show as a table']]", {
+          visible: true
+        });
+
+        await page.waitFor(2000);
+        await tableButton.click();
+
+        await page.waitForXPath("//div[@aria-label='Grid']");
+        return page.content();
+      };
+
+      // The data is timeseries but only for a limited timespan (a few
+      // weeks), so we'll cache it like regular data.
+      let scrapeDate = process.env.SCRAPE_DATE ? new Date(`${process.env.SCRAPE_DATE} 12:00:00`) : new Date();
+
+      const rawdata = await fetch.headless(this, this.url, 'default', scrapeDate, { callback });
+
+      const data = getDataFromPivotTable(rawdata);
+
+      const outData = {};
+      for (let ix = 0; ix < data.dates.length; ix++) {
```

Comment:
Why not just do a for each?

Reply:
... because?  No reason, other than that this is actually using that index to lookup three independent arrays.  forEach can handle that too, but this works.